src/platforms/hosted/jlink.c

```diff
@@ -364,11 +368,20 @@ bool jlink_select_interface(const uint8_t interface)
 static bool jlink_get_interfaces(void)
 {
 	uint8_t buffer[4U];
-	if (!jlink_simple_request_8(JLINK_CMD_INTERFACE_GET, JLINK_INTERFACE_GET_AVAILABLE, buffer, sizeof(buffer)))
-		return false;
+	/*
+	 * HW V5.40 IAR-KS 2006 hangs on 0xc7 commands (0xff and 0xfe),
+	 * and is known to not implement SWD or anything else besides JTAG.
+	 * Check the dedicated capability bit
+	 */
+	if (!(jlink.capabilities & JLINK_CAPABILITY_INTERFACES))
+		jlink.available_interfaces = JLINK_INTERFACE_AVAILABLE(JLINK_INTERFACE_JTAG);
```

Comment:
I would do an early return here instead of branching with the else for the remainder

Reply:
Any early return would bypass the selected interface logic below and the resulting print-out of the interfaces. We think that the logic in this PR is fine based on that reading.

Reply:
My bad, you're right. I didn't parse what i was reading correctly, quirks of the GH mobile interface...

LGTM